Common/src/interpolation_structure.cpp

```diff
@@ -2719,3 +2729,1240 @@ bool CSlidingMesh::CheckPointInsideTriangle(su2double* Point, su2double* T1, su2
+  
+  delete PPT;
+  delete [] coeff;
+  if (max_diff<max_diff_tol) {
+    delete [] write_row;
+    delete [] P_tmp;    
+  }
+  
+}
+
+void CRadialBasisFunction::Get_Distance(su2double *Coord_i, su2double *Coord_j, unsigned short nDim, su2double &dist)
+{
+  dist = 0;
+  for (unsigned short k=0; k<nDim; k++)
+    dist += pow((Coord_i[k] - Coord_j[k]), 2);
+  dist = sqrt(dist);
+}
+
+void CRadialBasisFunction::Get_RadialBasisValue(su2double &dist, CConfig *config)
+{
+  switch (config->GetKindRadialBasisFunction()) {
+
+    case WENDLAND_C2:
+      dist /= config->GetRadialBasisFunctionParameter();
+      if(dist < 1) dist = pow((1-dist), 4)*(4*dist+1);
+      else         dist = 0;
+      break;
+
+    case INV_MULTI_QUADRIC:
+      dist = 1/sqrt(dist*dist + \
+        (config->GetRadialBasisFunctionParameter())* \
+        (config->GetRadialBasisFunctionParameter()));
+      break;
+
+    case GAUSSIAN:
+      dist /= config->GetRadialBasisFunctionParameter();
+      dist = exp(-dist*dist);
+      break;
+			
+    case THIN_PLATE_SPLINE:
+      dist /= config->GetRadialBasisFunctionParameter();
+      if(dist > numeric_limits<float>::min())
+        dist *= dist*log(dist);
+      else
+        dist = 0;
+      break;
+		
+    case MULTI_QUADRIC:
+      dist = sqrt(dist*dist + \
+        (config->GetRadialBasisFunctionParameter())* \
+        (config->GetRadialBasisFunctionParameter()));
+      break;
+  }
+}
+
+/*--- Symmetric matrix class definitions ---*/
+CSymmetricMatrix::CSymmetricMatrix()
+{
+  initialized = false;
+  inversed    = false;
+  decomposed  = none;
+
+  val_vec       = NULL;
+  decompose_vec = NULL;
+  inv_val_vec   = NULL;
+  perm_vec      = NULL;
+}
+
+CSymmetricMatrix::~CSymmetricMatrix()
+{
+  if(val_vec)       {delete [] val_vec;}
+  if(decompose_vec) {delete [] decompose_vec;}
+  if(inv_val_vec)   {delete [] inv_val_vec;}
+  if(perm_vec)      {delete [] perm_vec;}
+}
+
+void CSymmetricMatrix::Initialize(int N)
+{
+  int i;
+
+  sz = N;
+  num_val = sz*(sz+1)/2;
+  val_vec = new double [num_val];
+  for (i=0; i<num_val; i++) {val_vec[i] = 0.0;}
+
+  initialized = true;
+}
+
+void CSymmetricMatrix::Initialize(int N, su2double *formed_val_vec)
+{
+  sz = N;	
+  num_val = sz*(sz+1)/2;
+
+  val_vec = new double [num_val];
+  for (int i=0; i<num_val; i++) {val_vec[i] = SU2_TYPE::GetValue(formed_val_vec[i]);}
+
+  initialized = true;
+}
+
+inline int CSymmetricMatrix::CalcIdx(int i, int j)
+{	
+  return max(i, j) + (2*sz-min(i, j)-1)*min(i, j)/2;
+}
+
+inline int CSymmetricMatrix::CalcIdxFull(int i, int j)
+{
+  return i*sz + j;
+}
+
+inline int CSymmetricMatrix::GetSize()
+{
+  return sz;
+}
+
+void CSymmetricMatrix::CholeskyDecompose(bool overwrite)
+{
+  int i, j, k;
+  double *vec, sum;
+	
+  if (!initialized) {
+    throw invalid_argument("Matrix not initialized.");
+  }
+
+  /*--- Point to correct vector ---*/
+  if (overwrite) {
+    vec = val_vec;
+  }
+  else {
+    decompose_vec = new double [num_val];
```

Comment:
If you have to use double here, use the typedef passivedouble. That eventually avoids some problems.

Reply:
Most of the code uses su2double but the areas that use LAPACK use double explicitly in case someone decides to make passivedouble a float (the code only calls the double version of the routines).